### PR TITLE
feat(credentials): document JMX credentials, keyring storage, header passthrough

### LINKED ISSUE #64: [Task] Add docs for matchExpression JMX credentials
https://github.com/cryostatio/cryostat/pull/1000
https://github.com/cryostatio/cryostat/pull/1019
https://github.com/cryostatio/cryostat-web/pull/481

### PR DESCRIPTION
**Pull request by andrewazores**
Fixes #64

### CODE REVIEW

File: guides/_subsections/configure-jmx-credentials-storage.md
Comment (maxcao13) [inline guides/_subsections/configure-jmx-credentials-storage.md:10]: Not sure, but I would change ex. to e.g.

File: guides/_subsections/configure-jmx-credentials-storage.md
Comment (maxcao13) [inline guides/_subsections/configure-jmx-credentials-storage.md:15]: `from the Cryostat server it displays` -> `from the Cryostat server, it displays`

File: guides/_subsections/configure-jmx-credentials-storage.md
Comment (maxcao13) [inline guides/_subsections/configure-jmx-credentials-storage.md:17]: `browser session then the` -> `browser session, then the`

File: guides/_subsections/configure-jmx-credentials-storage.md
Comment (maxcao13) [inline guides/_subsections/configure-jmx-credentials-storage.md:20]: `is present they supsersede` -> `is present, they supersede`

File: guides/_subsections/store-jmx-credentials.md
Comment (maxcao13) [inline guides/_subsections/store-jmx-credentials.md:4]: If you intend to use Automated Rules`,`

Comment (maxcao13) [inline guides/_subsections/store-jmx-credentials.md:4]: Even if you do not use Automated Rules`,` you may find it ....

I would also consider ending the sentence on `credentials in the keyring`. `This way, the Cryostat web UI does not need...`

File: guides/_subsections/store-jmx-credentials.md
Comment (maxcao13) [inline guides/_subsections/store-jmx-credentials.md:34]: If you have configured Backend JMX Credentials`,` then ...

**Comment by maxcao13** [comment]
I hope this is not actually being edited later and I did unnecessary work...

**Comment by andrewazores** [comment]
No, there would be no later editing stage for these upstream docs. We only have those resources for downstream product documentation and Red Hat blog posts.

**Review by maxcao13 — APPROVED** [review]
Looks good!

**Review by tthvo — APPROVED** [review]
Looks good! Also, looks like we have a new tab for Agent view coming up. Should we hold off until its merged and update the screenshots?

**Comment by andrewazores** [comment]
Most of the open PRs now are to merge into the upstream `version-220` branch, not straight into `main`. `version-220` will merge into `main` in a few weeks when we publish the upstream container images. I think I will start merging those PRs into `version-220` so we don't run into a bunch of conflicts and requiring work to update screenshots across different open PRs - the screenshot updates can all be done as a single PR against `version-220` later on, when that JMC Agent Plugin work is done and merged.

**Comment by tthvo** [comment]
Right thanks! That makes sense :D

**Pull request merged**
